feature/experimental/bgp/otg_tests/bgp_always_compare_med/bgp_always_compare_med_test.go

```diff
@@ -0,0 +1,583 @@
+	nbrs := []*bgpNeighbor{nbr1v4, nbr2v4, nbr3v4}
+
+	dutOcRoot := &oc.Root{}
+	ni1 := dutOcRoot.GetOrCreateNetworkInstance(deviations.DefaultNetworkInstance(dut))
+	ni_proto := ni1.GetOrCreateProtocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP")
+	bgp := ni_proto.GetOrCreateBgp()
+	global := bgp.GetOrCreateGlobal()
+	global.RouterId = ygot.String(dutDst2.IPv4)
+	global.As = ygot.Uint32(localAs)
+	global.GetOrCreateAfiSafi(oc.BgpTypes_AFI_SAFI_TYPE_IPV4_UNICAST).Enabled = ygot.Bool(true)
+
+	// Note: we have to define the peer group even if we aren't setting any policy because it's
+	// invalid OC for the neighbor to be part of a peer group that doesn't exist.
+	pg1 := bgp.GetOrCreatePeerGroup(peerGrpName1)
+	pg1.PeerAs = ygot.Uint32(ateAS1)
+	pg1.PeerGroupName = ygot.String(peerGrpName1)
+
+	pg2 := bgp.GetOrCreatePeerGroup(peerGrpName2)
+	pg2.PeerAs = ygot.Uint32(ateAS2)
+	pg2.PeerGroupName = ygot.String(peerGrpName2)
+
+	pg3 := bgp.GetOrCreatePeerGroup(peerGrpName3)
+	pg3.PeerAs = ygot.Uint32(ateAS3)
+	pg3.PeerGroupName = ygot.String(peerGrpName3)
+
+	for _, nbr := range nbrs {
+		nv4 := bgp.GetOrCreateNeighbor(nbr.neighborip)
+		nv4.PeerGroup = ygot.String(nbr.peerGrp)
+		nv4.PeerAs = ygot.Uint32(nbr.as)
+		nv4.Enabled = ygot.Bool(true)
+		af4 := nv4.GetOrCreateAfiSafi(oc.BgpTypes_AFI_SAFI_TYPE_IPV4_UNICAST)
+		af4.Enabled = ygot.Bool(true)
+		af6 := nv4.GetOrCreateAfiSafi(oc.BgpTypes_AFI_SAFI_TYPE_IPV6_UNICAST)
+		af6.Enabled = ygot.Bool(false)
+	}
+	return ni_proto
+}
+
+func sendTraffic(t *testing.T, otg *otg.OTG, c gosnappi.Config) {
+	t.Logf("Starting traffic")
+	otg.StartTraffic(t)
+	time.Sleep(trafficDuration)
+	t.Logf("Stop traffic")
+	otg.StopTraffic(t)
+}
+
+func verifyOTGBGPTelemetry(t *testing.T, otg *otg.OTG, c gosnappi.Config, state string) {
+	for _, d := range c.Devices().Items() {
+		for _, ip := range d.Bgp().Ipv4Interfaces().Items() {
+			for _, configPeer := range ip.Peers().Items() {
+				nbrPath := gnmi.OTG().BgpPeer(configPeer.Name())
+				_, ok := gnmi.Watch(t, otg, nbrPath.SessionState().State(), time.Minute, func(val *ygnmi.Value[otgtelemetry.E_BgpPeer_SessionState]) bool {
+					currState, ok := val.Val()
+					return ok && currState.String() == state
+				}).Await(t)
+				if !ok {
+					fptest.LogQuery(t, "BGP reported state", nbrPath.State(), gnmi.Get(t, otg, nbrPath.State()))
+					t.Errorf("No BGP neighbor formed for peer %s", configPeer.Name())
+				}
+			}
+		}
+		for _, ip := range d.Bgp().Ipv6Interfaces().Items() {
+			for _, configPeer := range ip.Peers().Items() {
+				nbrPath := gnmi.OTG().BgpPeer(configPeer.Name())
+				_, ok := gnmi.Watch(t, otg, nbrPath.SessionState().State(), time.Minute, func(val *ygnmi.Value[otgtelemetry.E_BgpPeer_SessionState]) bool {
+					currState, ok := val.Val()
+					return ok && currState.String() == state
+				}).Await(t)
+				if !ok {
+					fptest.LogQuery(t, "BGP reported state", nbrPath.State(), gnmi.Get(t, otg, nbrPath.State()))
```

Comment:
same as above, the first value of Watch contains the last value seen

Reply:
updated